### PR TITLE
fix real precision use for inverse and hyperbolic trig, scalbn, fdim, fma, pow

### PR DESCRIPTION
**Pull request by thewilsonator**
part of the fix for https://issues.dlang.org/show_bug.cgi?id=18559
Previous PR: https://github.com/dlang/phobos/pull/7463

### CODE REVIEW

File: std/math.d [hunk 1]
```diff
@@ -1322,39 +1322,45 @@ float cosh(float x) @safe pure nothrow @nogc  { return cosh(cast(real) x); }
...
+    import std.meta : AliasSeq;
+    static foreach (F; AliasSeq!(float, double, real))
+    {
+        assert(isIdentical(sinh(F(0.0)), F(0.0)));
+        assert(sinh(F(1.0)).approxEqual(F(sinh1)));
+    }
 }
 
-@safe @nogc nothrow unittest
+private F _sinh(F)(F x)
```
Comment (ibuclaw) [inline std/math.d:1345]: Similar functions have just gone with the naming convention `sinhImpl`, but not bothered either way.

**Review by ibuclaw — APPROVED** [review]

File: std/math.d [hunk 2]
```diff
@@ -4010,35 +4022,47 @@ real modf(real x, ref real i) @trusted nothrow @nogc
  *      $(TR $(TD $(PLUSMN)0.0)      $(TD $(PLUSMN)0.0) )
  *      )
  */
-real scalbn(real x, int n) @safe pure nothrow @nogc
+real scalbn(real x, int n) @safe pure nothrow @nogc { return _scalbn(x,n); }
+
+///
+double scalbn(double x, int n) @safe pure nothrow @nogc { return _scalbn(x,n); }
```
Comment (thewilsonator) [inline std/math.d:4028]:
```
std/math.d(4025:6)[warn]: Public declaration 'scalbn' has no documented example.
std/math.d(4028:8)[warn]: Public declaration 'scalbn' has no documented example.
```
Huh?

File: std/math.d [hunk 3]
```diff
@@ -4010,35 +4022,47 @@ real modf(real x, ref real i) @trusted nothrow @nogc
...
+///
+float scalbn(float x, int n) @safe pure nothrow @nogc { return _scalbn(x,n); }
+
+///
+@safe pure nothrow @nogc unittest
+{
+    assert(scalbn(0x1.2345678abcdefp0L, 999) == 0x1.2345678abcdefp999L);
+    assert(scalbn(-real.infinity, 5) == -real.infinity);
+    assert(scalbn(2.0,10) == 2048.0);
+    assert(scalbn(2048.0f,-10) == 2.0f);
```
Comment (thewilsonator) [inline std/math.d:4039]: Why do these not count?

File: std/math.d (hunk 2 above)
Comment (wilzbach) [reply std/math.d:4028]: You need to use `ditto` to group functions.

**Review by wilzbach — COMMENTED** [review]
-

File: std/math.d (hunk 2 above)
Comment (thewilsonator) [reply std/math.d]: Ahh, thanks!

**Pull request merged**